### ISSUE TITLE
[Feature] Add "click to play" option for all Video

### ISSUE THREAD

**Issue by donateur** [brave/brave-browser#11247]
## Description 
This is a feature request.

Websites like Ars Technica and Wired have videos embedded or immediately after their articles.  While they do not play audio by default, they download and play video content without prompting the user.  As someone on a restrictive data plan (on my smartphone) I do not want any video to play unless I click it first.

This feature exists in Firefox, but doesn't seem to exist in Chromium based browsers.

## Steps to Reproduce

   1.  Open and article like https://www.wired.com/story/ad-id-ios-android-tracking/ in brave
   2.  Scroll to the bottom of the page.
   3.  Observe the video playing (without sound enabled by default)

Interestingly Ars Technica only seem to have the embedded videos in the middle of content when rendered for Smartphones!

## Expected result:
Brave provides a new Setting which makes all video media autoplay.

## Desktop Brave version:
1.12.112 Chromium: 84.0.4147.125 (Official Build) (64-bit)
(OS is not relevant as this is a feature request)

## Android Device details:
   - Install type (ARM, x86): ARM
   - Device type (Phone, Tablet, Phablet): N/A Common to all
   - Android version: Android 10

## Other Additional Information:
Brave Shields do not appear to block this content.  However even if they did, they don't present the user with a visible video which they can choose to play by clicking on.

**Comment by tortino** [comment]
yes, please... ArsTechnica is eroding my data plan.

**Comment by donateur** [comment]
> yes, please... ArsTechnica is eroding my data plan.

I bet there are many more sites which do the same thing!  These are just the two which I read regularly and have noticed have automatically playing videos.

**Comment by pidus** [comment]
Loads of sites do this. On desktop I have to use an HTML5 video blocker extension. But mobile platform remains vulnerable.

**Comment by donateur** [comment]
> Loads of sites do this. On desktop I have to use an HTML5 video blocker extension. But mobile platform remains vulnerable.

I've proposed this as a feature for both the Desktop and Mobile (Android) versions, but you're right that the problem is worse on Android.  When using Firefox + uBlock origin, I could modify options in uBlock.  I don't have that option in Brave.

**Comment by TheHokieCoder** [comment]
A few notes based on my opinion of the over-arching issue that this feature request is based on:
- the Brave mobile browser on my Android device (v1.12.113 on Android 10) has a Control setting for "Background video playback", but this setting, currently set to `Disabled`, still allows the aforementioned videos on ArsTechnica and other sites to play automatically without any interaction from me. What "background video playback" is this setting blocking if it is not blocking the videos on ArsTechnica articles?
- if a setting were properly implemented in the browser to prevent **ALL** videos from auto-playing, this feature request could very well just simply be the "play" triangle overlay on the video, indicating to the user that playback will start when the user clicks/taps on the video.
- the argument about wasted data should not just apply to mobile and capped data plans. The user should have full control over what a website does, especially when it comes to multimedia. Even with an extremely fast, unlimited data plan I don't want any audio or video to automatically start playing without my consent.
- I'm actually quite surprised that a browser with so many features designed to protect the user from the web is still allowing videos to autoplay. If this doesn't get fixed soon, I'll probably switch to another mobile browser where I can control what happens when I browse. I am exhausted having to play whack-a-mole on my mobile device with the many websites that autoplay videos these days.

**Comment by donateur** [comment]
> A few notes based on my opinion of the over-arching issue that this feature request is based on:
> 
> * the Brave mobile browser on my Android device (v1.12.113 on Android 10) has a Control setting for "Background video playback", but this setting, currently set to `Disabled`, still allows the aforementioned videos on ArsTechnica and other sites to play automatically without any interaction from me. What "background video playback" is this setting blocking if it is not blocking the videos on ArsTechnica articles?
> * if a setting were properly implemented in the browser to prevent **ALL** videos from auto-playing, this feature request could very well just simply be the "play" triangle overlay on the video, indicating to the user that playback will start when the user clicks/taps on the video.
> * the argument about wasted data should not just apply to mobile and capped data plans. The user should have full control over what a website does, especially when it comes to multimedia. Even with an extremely fast, unlimited data plan I don't want any audio or video to automatically start playing without my consent.
> * I'm actually quite surprised that a browser with so many features designed to protect the user from the web is still allowing videos to autoplay. If this doesn't get fixed soon, I'll probably switch to another mobile browser where I can control what happens when I browse. I am exhausted having to play whack-a-mole on my mobile device with the many websites that autoplay videos these days.

Well said!

**Comment by TheHokieCoder** [comment]
@donateur Thanks for the clarification. Unfortunately my email from GitHub cut off after you said "Well said!", so I missed everything after that until I came back to post just now.

Unfortunately, the lack of communication about this feature request is telling to me that the maintainers are not concerned about protecting their users from _everything_ bad on the web. So I will be uninstalling Brave on all my devices. I'm just tired of having to pause videos on so many sites. I will be switching over to Firefox, which doesn't have this issue on mobile.

**Comment by donateur** [comment]
@TheHokieCoder it is disappointing to have no response.
I raised [another feature request back at the end of July](https://github.com/brave/brave-browser/issues/11033), which was immediately picked up and assigned to someone, yet that's had zero movement too.

I guess they have a small team - still it's really frustrating to have a generally excellent ad-blocker but with no option to tweak it to prevent video playback.  I too may use Firefox again soon, simply because uBlock origin will allow me to block the whole embedded videos I never watch.

**Comment by qhaas** [comment]
The [upstream community attempted](https://bugs.chromium.org/p/chromium/issues/detail?id=1063649#c8) requested this and was denied citing animated gifs would simply take their place.

However, autoplay goes beyond mere annoyances, data caps, and metered connections... it is actually an accessibility issue, as [Google pointed out when they originally blocked autoplay](https://developers.google.com/web/updates/2016/07/autoplay#the_finer_points).

**Comment by TheHokieCoder** [comment]
The argument that "bad actors" would switch to animated images or other alternatives to auto-playing videos is, again, a decision the user of the browser should be able to make. I don't believe any of us here are asking to make the _default_ that Brave blocks the autoplay of videos, but at least give us the option! It is annoying that Google, Brave developers, whomever are making decisions that _they_ think are best for us users. Give me the option to browse the web in the way that I see fits my needs and preferences.

To further the argument, if I were given the option to block the autoplaying of videos in my browser, and a certain website replaces it with an animated GIF in order to force me to watch their content, then I should be able to make the decision to never visit that site again as I would feel that website is going too far to shove content in my face or otherwise sell me on something. But, as an example, I don't feel that ArsTechnica is shoving their agenda or ads down my throat by autoplaying videos...they just see it as an available option and are using it. But I don't want that, and I should be able to disable it!

Why I am switching to Firefox is this simple reason that they give me the option to control this. There are a few things I have not liked about Firefox's UI and layout, compared to Chrome/Brave, but those just got bumped to a much lower priority because the Chromium project is making decisions for me with no regard to my preference.

**Comment by cjwijtmans** [comment]
> The [upstream community requested](https://bugs.chromium.org/p/chromium/issues/detail?id=1063649#c8) this and was denied citing animated gifs would simply take their place.
> 
> However, autoplay goes beyond mere annoyances, data caps, and metered connections... it is actually an accessibility issue, as [Google pointed out when they originally blocked autoplay](https://developers.google.com/web/updates/2016/07/autoplay#the_finer_points).

Then extend autoplay to disabling gifs playing automatically as well. Problem solved. I find it highly annoying when stuff is animating on the web without my consent. Audio. Video etc.

**Comment by qhaas** [comment]
> Then extend autoplay to disabling gifs playing automatically as well. Problem solved.

I agree.  They could also address problematic advertisers via their [coalition for better ads](https://blog.chromium.org/2020/02/videos-with-fewer-intrusive-ads.html).  Given chromium based browsers don't seem to be able to block (all) autoplay nearly as well as gecko based browsers, I'm thinking a change is needed upstream to do this 'right'.

The more I read about how autoplay video impacts those with certain cognitive disabilities, the more I'm convinced that full   autoplay blocking is a necessary accessibility feature.  I'm no lawyer or psychologist, but some of the stuff I'm reading implies this is almost 'digital' ADA territory.
